### ISSUE TITLE
single letter word removed when numbering

### ISSUE THREAD

**Issue by josemalonso** [jordan2175/markdown-tools#7]
Hi there, great tool so far; thanks for producing it and making it available.
I have noticed that if I start a title with a "one letter word" it is removed on numbering. for example, I have a heading "G Suite" and when numbering the numbering is applied correctly but the "G" disappears so the title becomes "X Suite" X being the heading number.
Thanks!

**Comment by jordan2175** [comment]
Thanks for the note. Yes, this is known issue with single letters starting a heading. I am thinking that the best way of dealing with this is to remove the option to do heading numbers with letters. I added that feature in haste, without really thinking it through.

**Comment by josemalonso** [comment]
Thanks. I see. I have only used headings with numbers so far but I would
guess there might be others using the heading numbers letters so adding the
feature might cause more harm than good? I mean, it's a minor issue I can
live with :-)

**Comment by jordan2175** [comment]
I am thinking of releasing a v2 of this plugin that will have some cool features like the ability to skip certain heading names and not number certain heading numbers etc.  In this version I will remove the ability to use letters for heading numbers, since I am not sure people use it.  But this will be Markdown-Tools-v2 so it will not impact any users of the current tool.  Let me know if something like this would work for you and if you have any features you would like to see.

**Comment by josemalonso** [comment]
Again, no strong opinion myself as I haven't faced such use case. Skipping
would be good. I wish Google would also do so in the ToC of the GDocs. I
have upvoted the request for them to implement.

**Comment by BenceSzalai** [comment]
A possible workaround is to use Punctuation space in headings after single letters at the beginning. The AddOn does not treats that character as a space so it thinks it is not a single letter.

In fact it'd be more convenient with the [no-break-space](https://en.wikipedia.org/wiki/Non-breaking_space) character which is more widely known, but unfortunately that one is treated just like a regular space.

So for now I'm using this character: " " (You can copy from here!) in my headings when my heading starts with a single letter, which in fact happens a lot as "a" is the definite grammatical article in Hungarian and used a lot as the first word of titles. But I think also in English it may happen fairly often to have a heading that starts with "a", as the indefinite grammatical article.

So I'll just leave it here for other folks facing the same issue: Use [unicode 2008 punctuation space](https://www.compart.com/en/unicode/U+2008) in the headings and it'll ensure the A (or other single letters) are not removed.

**Comment by josemalonso** [comment]
Interesting one! Thanks for looking into this again.

**Comment by jordan2175** [comment]
Yes, Hungarian (Magyarul) does use a lot of "a".  I like the idea of using that special space character.  I should try and add that to the code so it does that automatically. I have been working on a newer version of the add-on with a lot more features.
